src/megaclient.cpp

```diff
@@ -13520,6 +13527,22 @@ void MegaClient::putnodes_syncdebris_result(error, NewNode* nn)
 }
 #endif
 
+transfer_map::iterator MegaClient::getTransferByFileFingerprint(FileFingerprint *f, direction_t d, bool foreign, bool cached)
+{
+    transfer_map &auxTransfers = cached ? cachedtransfers[d] : transfers[d];
```

Comment:
Agree. Also, I thought about passing the map by reference, rather than a flag `cached`, so the iterator would always be for the right map. I'll change it.

Reply:
This function have a param cached and only if we want to find in cachedTransfers it's true so there's no problem because this param it's set depending on the map we want to compare after the call.

Reply:
Actually, if the caller requests a transfer from `cachedtransfers` and later on it compares the returned iterator against `transfers`, it would be a caller's mistake. The returned iterator is also used in the caller's code, so I'll change the `cached` param by a reference to the corresponding map :)

Reply:
Done! Please, if the change solve the issue, _Resolve_ this conversation. Thanks

Reply:
For me it's solved

Reply:
looks good, thanks, will resolve the conversation.